modules/programs/screen.nix

```diff
@@ -0,0 +1,29 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.screen;
+
+in
+{
+  options = {
+    programs.screen = {
+      enable = mkEnableOption "GNU screen";
+
+      settings = mkOption {
+        type = types.string;
+        default = {};
+        description = ''
+          Configuration file written to
+          <filename>~/.screenrc</filename>
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.screen ];
+    home.file.".screenrc".text = config.programs.screen.settings;
```

Comment:
Often it's nice to not create a configuration file if the user hasn't provided any configuration. So here you could have something like
```
home.file. = {
  ".screenrc".text = mkIf (cfg.settings != "") cfg.settings;
};
```